source/grid/grid_generator.cc

```diff
@@ -856,26 +856,44 @@ namespace GridGenerator
                              const std::vector<unsigned int> &subdivisions,
                              const bool colorize)
   {
-    if (subdivisions.size()==0)
+    // The provided edges and subdivisions may not be valid, so possibly
+    // recompute both of them.
+    std_cxx11::array<Tensor<1, spacedim>, dim> compute_edges = edges;
+    std::vector<unsigned int> compute_subdivisions = subdivisions;
+    if (compute_subdivisions.size() == 0)
       {
-        std::vector<unsigned int> new_subdivisions(dim, 1);
-        subdivided_parallelepiped<dim,spacedim>(tria, origin, edges, new_subdivisions, colorize);
-        return;
+        compute_subdivisions.resize(dim);
+        std::fill(compute_subdivisions.begin(), compute_subdivisions.end(), 1);
```

Comment:
Can't you simply write `compute_subdivisions.resize(dim,1)`, then you don't need to `fill` afterwards?